sql/filesort.cc

```diff
@@ -1160,7 +1160,8 @@ write_keys(Sort_param *param, Filesort_info *fs_info, uint count,
       DBUG_RETURN(1);                           /* purecov: inspected */
   }
 
-  if (my_b_write(chunk_file, &merge_chunk, sizeof(merge_chunk)))
+  if (my_b_write(chunk_file, reinterpret_cast<uchar*>(&merge_chunk),
```

Comment:
The same: `Merge_chunk*` and `uchar*` are considered to be `unrelated types`, so reinterpret_cast only